extension/content/tests/components/recipes/RecipeListing.test.tsx

```diff
@@ -0,0 +1,80 @@
+import { cleanup } from "@testing-library/react";
+import React from "react";
+
+import RecipeListing from "devtools/components/recipes/RecipeListing";
+
+import {
+  approvalRequestFactory,
+  revisionFactory,
+  recipeFactory,
+} from "devtools/tests/factories/recipes";
+
+afterEach(async () => {
+  jest.clearAllMocks();
+  await cleanup();
+});
+
+describe("RecipeListing", () => {
+  it("should have pending review tag when no approval", () => {
+    const approvalRequest = approvalRequestFactory.build({ approved: null });
+    const revision = revisionFactory.build(
+      { approval_request: approvalRequest },
+      { actionName: "branched-addon-study" },
+    );
+    const recipe = recipeFactory.build();
+    recipe.latest_revision = revision;
+
+    /* global renderWithContext */
+    // @ts-ignore
+    const { getByText } = renderWithContext(
+      <RecipeListing
+        copyRecipeToArbitrary={() => {}}
+        environmentName="prod"
+        recipe={recipe}
+      />,
+    );
+    expect(getByText("Pending Review")).toBeInTheDocument();
+  });
+
+  it("should have pending review tag when recipe rejected", () => {
```

Comment:
Should the description here be "should not have pending review tag..."?

Reply:
gah some bad c+p, lemme fix that